configure

```diff
@@ -2147,9 +2210,9 @@ sub write_environment {
   print $SE "$specific{'comment'} OpenDDS configure script: $0 $args_\n";
   print $MK "$specific{'comment'} OpenDDS configure script: $0 $args_\n" if $MK;
 
-  for my $key (keys %buildEnv) {
+  for my $key (sort(keys(%buildEnv))) {
     if ($key ne 'build' && $key ne 'mpctype') {
-      print $SE "$specific{'export'} $key=$buildEnv{$key}\n";
+      print $SE "$specific{'export'} $key=\"$buildEnv{$key}\"\n";
```

Comment:
Is this quoting going to work in all cases?  I don't remember if `%buildEnv` can come from the user's command-line options.  If so, could it already have a quote? Does the makefile below `$MK` also need quoting?

Reply:
> Is this quoting going to work in all cases? 

Maybe not, as this might be causing the missing Perl on the Windows builds.

>I don't remember if `%buildEnv` can come from the user's command-line options. If so, could it already have a quote?

I don't think they should already be quoted, but I probably have to look into that more.

> Does the makefile below` $MK` also need quoting?

Makefiles apparently doesn't need quoting:

```shell
$ echo -n "var=Hello goodbye\n\$(info \$(var))\n" | make -f -
Hello goodbye
make: *** No targets.  Stop.
```

Reply:
> I don't remember if %buildEnv can come from the user's command-line options. If so, could it already have a quote?

Okay I played around with this a little. In batch script I don't think the behavior will break anything if I do this correctly using `set "NAME=VALUE"` as I just pushed. It seems to ignore all quotes expect for the last one if the name starts with a quote.

```batch
>set foo="bar
>echo %foo%
"bar
>set "foo="bar"
>echo %foo%
"bar

>set foo="bar""
>echo %foo%
"bar""
>set "foo="bar"""
>echo %foo%
"bar""
```

I think spaces already work, but quoting it allows it to contain special characters like `&`.

```cmd
>set foo= bar 
>echo {%foo%}
{ bar }

>set foo=bar&
>echo "%foo%"
"bar"
>set "foo=bar&"
>echo "%foo%"
"bar&"
```

Of course shell script is different. In the commit I just I pushed it's removing removing surrounding double quotes and escaping any other double quotes.